libcaf_core/caf/flow/observable.test.cpp

```diff
@@ -543,6 +543,40 @@ TEST("on_error_return() replaces an error with a value") {
   }
 }
 
+TEST("start_with(value) builds observable that emits value first") {
+  const auto value = 1;
+  SECTION("start_with(value) adds value as first element") {
```

Comment:
Please also add a test for passing an observable to `start_with`.

Reply:
I have already added the following test. Shall I move it to a different section?

```suggestion
  check_eq(collect(build(range(2, 3).start_with(range(1, 2)))),
               vector{1, 2, 2, 3, 4});
    }
```